### DISCUSSION TITLE
pandoc issue with `tar_render` when rendering site

### DISCUSSION THREAD

**Discussion by jennysjaarda** [ropensci/targets#648] [Ideas]
I am trying to use `tar_render` to render some markdown documents that are part of a website (`workflowr`, specifically) and I get an error (not `targets` specifc) when I attempt this: 

```
list(
... # more targets
 tar_render(rmd_index, "analysis/index.Rmd"),

)

tar_make(names = "rmd_index")

```

Error

```
pandoc: Could not determine mime type for `/data/sgg2/jenny/projects/proxyMR/renv/library/R-4.1/x86_64-pc-linux-gnu/rmarkdown/rmd/h/fontawesome/css/../webfonts/fa-brands-400.woff2' 
Error: pandoc document conversion failed with error 1
```

I get the same error when I run `rmarkdown::render("analysis/final_phenotype_list.Rmd")`, confirming it's a `render` specific error. But if I run `rmarkdown::render_site("analysis/final_phenotype_list.Rmd")` it works as expected (because I have `_site.yml` document). I really like the way `tar_render` looks for `tar_load` and `tar_read` functions within the document. I was wondering if you would consider adding a nearly identical function but `tar_render_site`, so I could incorporate my site documents into the targets pipeline? (I'm not ready to quite make the switch to Targets Markdown, it seems like a big undertaking.) 

Thanks in advance for your thoughts/ suggestions!

**Comment by wlandau** [comment]
I would rather stick with the more general `tar_render()` in `tarchetypes`. As a workaround, to invalidate the report when `_site.yml` changes, you can include `_site.yml` as an upstream file target and then pass the target name to `...` in `tar_render()`.

**Reply by jennysjaarda** [reply]
Hi Will, 
just another followup. there seems to be several features in `render_site()` that don't exist simply within `render()`. I'm not sure how best to use the `targets` features inside a site if I stick to `tar_render()`. Perhaps @jdblischak has some ideas, otherwise I think best to site docs outside of the targets pipeline (or stick to `wflow_build` or `render_site` with a pipeline, but lose the nice features of `tar_render` which looks for `tar_read` and `tar_load` functions).

**Comment by jennysjaarda** [comment]
Good idea to add `_site.yml` as an upstream file target so downstream markdown reports depend on it. However I'm still not sure how it will solve the issue exactly. Maybe I'm missing something, but I can't get `rmarkdown::render()` to work with a `_site.yml` file in the directory, I seem to need to use the `render_site()` function.

**Comment by jennysjaarda** [comment]
So I tried to run `rmarkdown::render("analysis/final_phenotype_list.Rmd")` on my personal computer and all works fine (except that the output is in the wrong folder, which may also be the case with `tar_render()`. But for some reason on linux I am getting an error with `render()` that I don't get with `render_site()` which is why I thought `targets` needs a function specific to `render_site()`. I can't seem to reproduce it, but I will work on that. The error is (just in case it is more meaningful to you than me) is below: 

```
pandoc: Could not determine mime type for `/data/sgg2/jenny/projects/proxyMR/renv/library/R-4.1/x86_64-pc-linux-gnu/rmarkdown/rmd/h/fontawesome/css/../webfonts/fa-brands-400.woff2'
Error: pandoc document conversion failed with error 1
```

**Reply by jennysjaarda** [reply]
Just to follow-up, the problem seems to be the following lines in `_site.yml`: 

```
  right:
  - icon: fa-github
    text: Source code
    href: https://github.com/jennysjaarda/PSYMETAB
```

specifically where it fetches the icon: `  - icon: fa-github`

If I remove this line, `render()` works as expected (on linux platform). I don't know if there's other features that are affected in `render` vs `render_site` in addition.

**Comment by ginolhac** [comment]
Hello,
I included the `_site.yml` as an upstream targets myself.

For `render_site()` I haven't figured out why, but I need to wrap the call to it in a `callr` usage such as:

```r
# knit site
knit_base <- function(Rmd, navbar = NULL, dir = "../docs") {
  tar_cancel(is.null(Rmd))
  Rmd <- here(Rmd)
  # rewrite output dir
  writeLines(sub("output_dir: \"docs\"",
                 paste0("output_dir: \"", dir, "\""),
                 readLines("site/_site.yml")), "site/_site.yml")
  callr::r(function(...) rmarkdown::render_site(...),
           args = list(input = Rmd,
                       quiet = TRUE))
}
```

`navbar` is an upstream target that create the navbar and is using the `_site.yml`
